WLogger/Repository/Commons/ImageFetcher.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  ImageFetcher.swift
+//  WLogger
+//
+//  Created by Pablo Giorgi on 1/28/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+
+public struct  ImageFetcher {
+    
+    public func fetchImage(imageURL: NSURL) -> SignalProducer<UIImage?, NSError> {
+        return NSURLSession.sharedSession().rac_dataWithRequest(NSURLRequest(URL: imageURL)).map { UIImage(data: $0.0) }
```

Comment:
In this case is OK to force unwrap the optional `UIImage!(data: $0.0)`  or you could do something like

```swift
enum ImageFetcherError {
case InvalidImageFormat
case FetchError(NSError)
}

public func fetchImage(imageURL, NSURL) -> SignalProducer<UIImage, ImageFetcherError> {
    return NSURLSession.sharedSession()
        .rac_dataWithRequest(NSURLRequest(URL: imageURL))
        .flatMapError { SignalProducer(error: .FetchError($0)) }
        .flatMap(.Concat) { data, _ in 
            if let image = UIImage(data: data) {
                return SignalProducer(value: image)
            } else { 
                return SignalProducer(error: .InvalidImageFormat)
            }
        }
}
```